frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx

```diff
@@ -0,0 +1,227 @@
+      createMockDashboardCard({
+        id: DASHCARD_BAR_ID,
+        dashboard_tab_id: TAB_ID,
+        card: createMockCard({ id: CARD_BAR_ID, name: "Bar", display: "bar" }),
+        size_x: 12,
+        size_y: 8,
+        parameter_mappings: [
+          {
+            card_id: CARD_BAR_ID,
+            parameter_id: PARAMETER_ID,
+            target: ["variable", ["template-tag", "abc"]],
+          },
+        ],
+      }),
+      createMockDashboardCard({
+        id: DASHCARD_TABLE_ID,
+        dashboard_tab_id: TAB_ID,
+        card: createMockCard({
+          id: CARD_TABLE_ID,
+          name: "Table",
+          display: "table",
+        }),
+        ...(!hasScroll ? { col: 12 } : {}),
+        size_x: 12,
+        size_y: 8,
+      }),
+    ],
+  });
+}
+
+const Template: ComponentStory<typeof PublicOrEmbeddedDashboardView> = args => {
+  return <PublicOrEmbeddedDashboardView {...args} />;
+};
+
+const defaultArgs: Partial<
+  ComponentProps<typeof PublicOrEmbeddedDashboardView>
+> = {
+  dashboard: createDashboard(),
+  titled: true,
+  bordered: true,
+  background: true,
+  slowCards: {},
+  selectedTabId: TAB_ID,
+  parameters: [
+    createMockParameter({
+      id: PARAMETER_ID,
+    }),
+  ],
+};
+
+// Light theme
+export const LightThemeDefault = Template.bind({});
+LightThemeDefault.args = defaultArgs;
+
+export const LightThemeScroll = Template.bind({});
+LightThemeScroll.args = {
+  ...defaultArgs,
+  dashboard: createDashboard({ hasScroll: true }),
+};
+LightThemeScroll.decorators = [ScrollDecorator];
+
+export const LightThemeNoBackgroundDefault = Template.bind({});
+LightThemeNoBackgroundDefault.args = {
+  ...defaultArgs,
+  background: false,
+};
+
+export const LightThemeNoBackgroundScroll = Template.bind({});
+LightThemeNoBackgroundScroll.args = {
+  ...defaultArgs,
+  background: false,
+  dashboard: createDashboard({ hasScroll: true }),
+};
+LightThemeNoBackgroundScroll.decorators = [ScrollDecorator];
+
+// Dark theme
+export const DarkThemeDefault = Template.bind({});
+DarkThemeDefault.args = {
+  ...defaultArgs,
+  theme: "night",
+};
+
+export const DarkThemeScroll = Template.bind({});
+DarkThemeScroll.args = {
+  ...defaultArgs,
+  theme: "night",
+  dashboard: createDashboard({ hasScroll: true }),
+};
+DarkThemeScroll.decorators = [ScrollDecorator];
+
+export const DarkThemeNoBackgroundDefault = Template.bind({});
+DarkThemeNoBackgroundDefault.args = {
+  ...defaultArgs,
+  theme: "night",
+  background: false,
+};
+
+export const DarkThemeNoBackgroundScroll = Template.bind({});
+DarkThemeNoBackgroundScroll.args = {
+  ...defaultArgs,
+  theme: "night",
+  background: false,
+  dashboard: createDashboard({ hasScroll: true }),
+};
+DarkThemeNoBackgroundScroll.decorators = [ScrollDecorator];
+
+const EXPLICIT_SIZE_WAIT_TIME = 300;
+function ScrollDecorator(Story: Story) {
+  useEffect(() => {
+    setTimeout(() => {
+      document.querySelector("[data-testid=embed-frame]")?.scrollBy(0, 9999);
```

Comment:
Let me know if there's a better way to scroll the stories. I need to scroll to test sticky filters.

Reply:
Scrolling does not trigger the ExplicitSize, is that right? Looks more like a scroll_wait_time